File: app/javascript/channels/notif_channel.js
import consumer from "./consumer"

console.log("loading notif channel js");

consumer.subscriptions.create({channel: "NotifChannel", video_id: 1}, {
  connected() {
    // Called when the subscription is ready for use on the server
    console.log("cable connected");
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
  }
});

File: app/channels/notif_channel.rb
class NotifChannel < ApplicationCable::Channel
  def subscribed
    # stream_from "some_channel"
    puts "subbed video:#{params[:video_id]}"
    stream_from "video:#{params[:video_id]}"
  end

  def receive(data)
    # could be used if the frontend needs to send a msg out directly
    # currently the server is generating all cable msgs
    puts "cable data received"
    puts data
    #ActionCable.server.broadcast("video:#{Video.find(params[:video_id].url)}", data)
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end
end
